table/block_based/block_based_table_builder.cc

```diff
@@ -450,6 +460,106 @@ struct BlockBasedTableBuilder::Rep {
   ~Rep() {}
 };
 
+class BlockBasedTableBuilder::ParallelCompressionRep {
```

Comment:
`ParallelCompressionRep` was designed as a helper class for parallel compression in block based table only. `ParallelCompressionRep` will only be used by `BlockBasedTableBuilder`, so I made it a private inner class so that it's not visible to other code, including those who used `block_based_table_builder.h`.